recipes-utils/exfatprogs/exfatprogs/run-ptest

```diff
@@ -0,0 +1,71 @@
+#!/bin/sh
+
+rm -f ptest.log || true
+rm -f test_file_creation.log || true
+
+loopdev=$(losetup -f)
+
+mountdir="/mnt/exfatprogs_ptest"
+
+if [ ! -d "${mountdir}" ]; then
+  mkdir "${mountdir}"
+fi
+
+exfat_support=$(grep exfat /proc/filesystems >/dev/null ; echo $?)
+module_installed=0
+
+if [ "${exfat_support}" = "1" ]; then
+  insmod /lib/modules/$(uname -r)/kernel/fs/exfat/exfat.ko
+  if [ $? -ne 0 ]; then
+    echo "Error: This kernel is not supporting exfat filesystem."
+    exit 1
+  fi
+  module_installed=1
+fi
+
+cd tests
+./test_fsck.sh 2>&1 | sed -u -e '/^Passed \.\//s/^Passed \(.*\)/PASS: \1/g' -e '/^Failed/s/^Failed \(.*\)/FAIL: \1/g' | tee ../ptest.log
```

Comment:
Could you update "Test result"?
I'm concerned about whether "Passed X of Y" will increase the number of "Passed: X" in the result.
At least the previous output seems to have this problem.

```
Passed 17 of 17
... <snip> ...
Test Result
Total tests: 18
Passed: 18
```

Reply:
Current code doesn't show "Test Result" lines.

There are 2 test script the one is  [test_fsck.sh](https://github.com/exfatprogs/exfatprogs/blob/1.2.5/tests/test_fsck.sh)  which currently runs 17 test cases. shows So that it shows "Passed 17 of 17".
The other  script test_file_creation.sh  runs 1 test. so total test num beris 18. That's why "Total tests: 18" was outputted.

Reply:
I updated test result output in PR comment.

Reply:
Note: Ptest results logs should be better to be compatible with the TAP standard [tap](https://testanything.org/)

Reply:
@masami256 
> I updated test result output in PR comment.

Shouldn't "ERROR:" be changed to "FAIL:"?

Reply:
@teradat no.

These messages are shown by fsck.exfat tries to repair invalid/corrupted file image, not test result. The test_fsck.sh tests invalid/corrupted file images.   Therefore, such error messages  shouldn't be modified or count as failed tests.

```
ERROR: the cluster chain of root is broken. Truncate (y/N)? y
ERROR: /: the checksum 0xdede of a file is wrong, expected: 0x793 at 0x203240. Delete (y/N)? y
ERROR: /: unknown entry type 0xc0 at 0x203260. Delete (y/N)? y
ERROR: /: unknown entry type 0xc1 at 0x203280. Delete (y/N)? y
ERROR: /: the checksum 0x828a of a file is wrong, expected: 0x51da at 0x21cfe0. Delete (y/N)? y
```

These tests expect fsck.exfat return error code so that you can see ___PASS: ./bad_root___  in the log.

```
Running ./bad_root
-----------------------------------
failed to follow the cluster chain of root
failed to verify root directory.
exfatprogs version : 1.2.5
ERROR: the cluster chain of root is broken. Truncate (y/N)? y
ERROR: /: the checksum 0xdede of a file is wrong, expected: 0x793 at 0x203240. Delete (y/N)? y
ERROR: /: unknown entry type 0xc0 at 0x203260. Delete (y/N)? y
ERROR: /: unknown entry type 0xc1 at 0x203280. Delete (y/N)? y
ERROR: /: the checksum 0x828a of a file is wrong, expected: 0x51da at 0x21cfe0. Delete (y/N)? y
exfat.img: clean. directories 3, files 87
exfat.img: files corrupted 0, files fixed 2

exfatprogs version : 1.2.5
exfat.img: clean. directories 4, files 88

PASS: ./bad_root
```